### ISSUE TITLE
Incorrect message when using a payment condition is within a restriction set

### ISSUE THREAD

**Issue by rjnl** [sebsoftnl/moodle-availability_gwpayments#7]
When a payment restriction is within a restriction set, we get the following message:
No gateway available for this activity. This is usually the case when the expected currency is not available for any gateway or one or more gateway(s) is/are disabled.

The message is incorrect because if a restriction set is not used, then everything works correctly.

If you turn on debugging, then you see this stack trace:

```
Notice:  Undefined property: stdClass::$type in  C:\Wamp\www\moodle\availability\condition\gwpayments\classes\payment\service_provider.php  on line 92
--


1 | 0.0012 | 462376 | {main}(  ) | ...\view.php:0
2 | 1.7128 | 15303608 | require( 'C:\Wamp\www\moodle\course\format\topics\format.php' ) | ...\view.php:278
3 | 1.7161 | 15302096 | format_topics_renderer->print_multiple_section_page(  ) | ...\format.php:57
4 | 1.8007 | 15327504 | core_course_renderer->course_section_cm_list(  ) | ...\renderer.php:859
5 | 1.9314 | 15995816 | core_course_renderer->course_section_cm_list_item(  ) | ...\renderer.php:1051
6 | 1.9314 | 15995816 | core_course_renderer->course_section_cm(  ) | ...\renderer.php:836
7 | 1.9679 | 16040336 | core_course_renderer->course_section_cm_availability(  ) | ...\renderer.php:965
8 | 1.9681 | 16040496 | core_availability\info_module->get_full_information(  ) | ...\renderer.php:810
9 | 1.9683 | 16042424 | core_availability\tree->get_full_information(  ) | ...\info.php:275
10 | 1.9683 | 16042424 | core_availability\tree->get_full_information_recursive(  ) | ...\tree.php:459
11 | 1.9683 | 16042824 | core_availability\tree->get_full_information_recursive(  ) | ...\tree.php:523
12 | 1.9683 | 16043224 | availability_gwpayments\condition->get_standalone_description(  ) | ...\tree.php:526
13 | 1.9693 | 16049232 | availability_gwpayments\condition->get_description(  ) | ...\condition.php:229
14 | 1.9693 | 16049232 | availability_gwpayments\condition->get_either_description(  ) | ...\condition.php:174
15 | 1.9705 | 16050272 | availability_gwpayments\condition->can_payment_be_made(  ) | ...\condition.php:204
16 | 1.9826 | 16182496 | core_payment\helper::get_available_gateways(  ) | ...\condition.php:271
17 | 1.9826 | 16182496 | core_payment\helper::get_payable(  ) | ...\helper.php:71
18 | 1.9854 | 16203752 | component_class_callback(  ) | ...\helper.php:200
19 | 1.9854 | 16203848 | availability_gwpayments\payment\service_provider::get_payable(  ) | ...\moodlelib.php:7992
```

The issue is caused in availability_gwpayments\payment\service_provider::get_payable()

**Comment by rogiervandongen** [comment]
@rjnl Was the (merged) PR fixing this issue? In that case this issue could be closed :)

**Comment by rjnl** [comment]
Yes, this was fixed with #8
